components/canvas/canvas_paint_thread.rs

```diff
@@ -547,6 +547,10 @@ impl<'a> CanvasPaintThread<'a> {
 
     fn recreate(&mut self, size: Size2D<i32>) {
         self.drawtarget = CanvasPaintThread::create(size);
+        // Webrender doesn't let images change size, so we clear the webrender image key.
+        if let Some(image_key) = self.image_key.take() {
+            self.webrender_api.delete_image(image_key);
+        }
```

Comment:
r=me, but could you do the same in the `webgl_paint_thread`? Right now the readback path suffers the same problem afaict.